parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,203 @@
+		SubstitutionSchedule substitutionSchedule = SubstitutionSchedule.fromData(scheduleData);
+		substitutionSchedule.setClasses(getAllClasses());
+		substitutionSchedule.setTeachers(getAllTeachers());
+		final JSONArray changes = getChanges();
+		final JSONArray courses = getCourses();
+		final JSONArray teachers = getTeachers();
+
+		parseLegionBoard(substitutionSchedule, changes, courses, teachers);
+		return substitutionSchedule;
+	}
+
+	/**
+	 * Returns authentication key as shown
+	 * <a href="https://gitlab.com/legionboard/heart/blob/master/doc/README.md">in the documentation</a>.
+	 */
+	private String getAuthenticationKey(Credential credential) {
+		final UserPasswordCredential userPasswordCredential = (UserPasswordCredential) credential;
+		final String username = userPasswordCredential.getUsername();
+		final String password = userPasswordCredential.getPassword();
+		return DigestUtils.sha256Hex(username.toLowerCase() + "//" + password);
+	}
+
+	/**
+	 * Returns a JSONArray with all changes from now to in one week.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/changes/list.md">List changes</a>
+	 */
+	private JSONArray getChanges() throws IOException, JSONException {
+		// Date (or alias of date) when the changes start
+		final String startBy = "now";
+		// Date (or alias of date) when the changes end
+		final String endBy = "i1w";
+
+		final String url = api + "/changes?startBy=" + startBy + "&endBy=" + endBy + "&k=" + getAuthenticationKey(getCredential());
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all courses.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/courses/list.md">List courses</a>
+	 */
+	private JSONArray getCourses() throws IOException, JSONException {
+		final String url = api + "/courses?k=" + getAuthenticationKey(getCredential());
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all teachers.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/teachers/list.md">List teachers</a>
+	 */
+	private JSONArray getTeachers() throws IOException, JSONException {
+		final String url = api + "/teachers?k=" + getAuthenticationKey(getCredential());
+		return getJSONArray(url);
+	}
+
+	private JSONArray getJSONArray(String url) throws IOException, JSONException, HttpResponseException {
+		try {
+			return new JSONArray(httpGet(url, "UTF-8"));
+		} catch (HttpResponseException httpResponseException) {
+			if (httpResponseException.getStatusCode() == 404) {
+				return null;
+			}
+			throw httpResponseException;
+		}
+	}
+
+	private void parseLegionBoard(SubstitutionSchedule substitutionSchedule, JSONArray changes, JSONArray courses, JSONArray teachers) throws IOException, JSONException {
+		if (changes == null) {
+			return;
+		}
+		// Link course IDs to their names
+		HashMap<String, String> coursesHashMap;
+		if (courses != null) {
+			coursesHashMap = new HashMap<>();
```

Comment:
Ah, danke für die Erklärung. Ich bin wohl schon zu verwöhnt von PHP bei Heart :D

On July 24, 2016 11:54:18 AM GMT+02:00, Johan von Forstner <notifications@github.com> wrote:
>> +		} catch (HttpResponseException httpResponseException) {
>> +			if (httpResponseException.getStatusCode() == 404) {
>> +				return null;
>> +			}
>> +			throw httpResponseException;
>> +		}
>> +	}
>> +
>> +	private void parseLegionBoard(SubstitutionSchedule
>substitutionSchedule, JSONArray changes, JSONArray courses, JSONArray
>teachers) throws IOException, JSONException {
>> +		if (changes == null) {
>> +			return;
>> +		}
>> +		// Link course IDs to their names
>> +		HashMap<String, String> coursesHashMap;
>> +		if (courses != null) {
>> +			coursesHashMap = new HashMap<>();
>
>Nein, so was gibt es nur bei primitiven Typen (bei `boolean` `false`
>und bei `int` `0` etc.). So gibt es halt einen Compiler-Fehler sobald
>du versuchst, an einer Stelle darauf zuzugreifen wo es eventuell nicht
>initialisiert wurde.
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/johan12345/substitution-schedule-parser/pull/6/files/87f679af0282a20db485917fde2201644a993d7e#r71985718